jwt/jwt.go

```diff
@@ -68,14 +68,19 @@ func Sign(req *http.Request, key *key.PrivateKey, params config.SignerParams) er
 	return nil
 }
 
-func Verify(req *http.Request, keyServer keyserver.Reader, nonceVerifier noncestorage.NonceStorage, expectedAudience string, maxSkew time.Duration, maxTTL time.Duration) (jose.Claims, error) {
-	// Extract token from cookie.
-	cookieExtractor := oidc.CookieTokenExtractor("access_token")
-	token, err := cookieExtractor(req)
-	if err != nil {
+func Verify(req *http.Request, keyServer keyserver.Reader, nonceVerifier noncestorage.NonceStorage, cookiesEnabled bool, expectedAudience string, maxSkew time.Duration, maxTTL time.Duration) (jose.Claims, error) {
+	var token = ""
+
+	// Extract token from cookie if enabled.
+	if cookiesEnabled {
+		cookieExtractor := oidc.CookieTokenExtractor("access_token")
+		token, _ = cookieExtractor(req)
+	}
+
+	if token == "" {
 		// Not found in cookie, extract from header
-		token, err = oidc.ExtractBearerToken(req)
-		if err != nil {
+		token, _ = oidc.ExtractBearerToken(req)
```

Comment:
Not sure if it's a good idea. Sice oids is an external lib, it is better to use it is a stardart black-box way - use error check instead of relying on implementation returning ""

Reply:
Could you please elaborate more why comparing with an empty string could not work if the error is not `nil`? ASAIU the only possible value for a string in go is an empty string or real value. Then empty string should be returned if an exception occurs or cookies contain empty value. Do you think we should not override empty token from cookies?

Reply:
it is not guarateed that `oidc.ExtractBearerToken` always will return empty string in case of error. There can be nested errors  like  exceptions in `strings.ToUpper` etc so it better to not rely on spcecific implementation.